### ISSUE TITLE
[QUESTION] mtp_loss is scaled incorrectly

### ISSUE THREAD

**Issue by bisunny** [NVIDIA/Megatron-LM#2417]
**Your question**
Ask a clear and concise question about Megatron-LM.
When `calculate_per_token_loss` is enabled, `finalize_model_grads` scales the gradients according to the num tokens（total number of a iter). However, I observed that the number of tokens (loss mask) corresponding to `mtp_loss` might be inconsistent with num_tokens in `finalize_model_grads`. I think there is a bug.

**Comment by BestJuly** [comment]
Thanks for reporting this issue. Yes, this is indeed a bug. 

The numerical impact per training step is relatively small for typical pre-training scenarios:
- With 1 MTP layer and seqlen=4096: ~0.024% under-weighting (1/4096)
- With 2 MTP layers: layer 1 ~0.024%, layer 2 ~0.049% 
- Impact is larger for shorter sequences (e.g., ~0.1% for seqlen=1024)

I will implemented a quick fix.

**Comment by janEbert** [comment]
Since #3159 is merged, which addresses the issue, I'm closing this. Feel free to re-open if something is still wrong.
